client.go

```diff
@@ -113,17 +131,18 @@ var DefaultDialer = &Dialer{
 // nilDialer is dialer to use when receiver is nil.
 var nilDialer Dialer = *DefaultDialer
 
-// Dial creates a new client connection. Use requestHeader to specify the
+// DialWithContext creates a new client connection. Use requestHeader to specify the
```

Comment:
Name the function DialContext.

Reply:
I think it should remain as `DialWithContext`, using the standard library as inspiration: https://golang.org/pkg/net/http/#Request.WithContext